services/motion/builtin/builtin.go

```diff
@@ -237,71 +247,170 @@ func (ms *builtIn) MoveOnGlobe(
+
+			inputs, err := kb.CurrentInputs(ctx)
+			if err != nil {
+				return false, nil
+			}
+			if len(kb.Kinematics().DoF()) == 2 {
+				inputs = inputs[:2]
+			}
+			planRequest.Inputs = map[string][]referenceframe.Input{componentName.Name: inputs}
+
+			plan, err := motionplan.PlanMotion(ctx, planRequest)
+			if err != nil {
+				return false, nil
+			}
+
+			// drain the replanChan
+			for len(replanChan) > 0 {
+				<-replanChan
+			}
+			// drain the replanChan
+			for len(errChan) > 0 {
+				<-errChan
+			}
+
+			if positionPollingPeriod > 0 {
+				startPolling(cancelCtx, positionPollingPeriod, func(ctx context.Context) error {
+					// TODO: the function that actually monitors position
+					fmt.Println("position poll")
+					return nil
+				})
+			}
+			if obstaclePollingPeriod > 0 {
+				startPolling(cancelCtx, obstaclePollingPeriod, func(ctx context.Context) error {
+					// TODO: the function that actually monitors obstacles
+					fmt.Println("obstacle poll")
+					replanChan <- true
+					return nil
+				})
+			}
+
+			backgroundWorkers.Add(1)
+			goutils.ManagedGo(func() {
+				if err := ms.executePlan(cancelCtx, kb, plan); err != nil {
+					errChan <- err
+					return
+				}
+
+				position, _, err := movementSensor.Position(cancelCtx, nil)
+				if err != nil {
+					errChan <- err
+					return
+				}
+
+				ms.logger.Infof("position: %v", position)
+				// TODO: dont do the calculation to mm here
+				if spatialmath.GeoPointToPose(position, destination).Point().Norm() <= 1e3*motionCfg.PlanDeviationM {
+					successChan <- true
+					return
+				}
+			}, backgroundWorkers.Done)
 		}
 	}
-	return true, nil
 }
 
-// planMoveOnGlobe returns the plan for MoveOnGlobe to execute.
-func (ms *builtIn) planMoveOnGlobe(
+func (ms *builtIn) newMoveOnGlobeRequest(
 	ctx context.Context,
 	componentName resource.Name,
 	destination *geo.Point,
-	movementSensorName resource.Name,
+	movementSensor movementsensor.MovementSensor,
 	obstacles []*spatialmath.GeoObstacle,
-	kinematicsOptions kinematicbase.Options,
+	motionCfg *motion.MotionConfiguration,
 	extra map[string]interface{},
-) ([][]referenceframe.Input, kinematicbase.KinematicBase, error) {
-	// build the localizer from the movement sensor
-	movementSensor, ok := ms.movementSensors[movementSensorName]
-	if !ok {
-		return nil, nil, resource.DependencyNotFoundError(movementSensorName)
+) (*motionplan.PlanRequest, kinematicbase.KinematicBase, error) {
+	kinematicsOptions := kinematicbase.NewKinematicBaseOptions()
+	if motionCfg.LinearMPerSec != 0 {
+		kinematicsOptions.LinearVelocityMMPerSec = motionCfg.LinearMPerSec * 1000
+	}
+	if motionCfg.AngularDegsPerSec != 0 {
+		kinematicsOptions.AngularVelocityDegsPerSec = motionCfg.AngularDegsPerSec
 	}
+	if motionCfg.PlanDeviationM != 0 {
+		kinematicsOptions.PlanDeviationThresholdMM = motionCfg.PlanDeviationM * 1000
+	}
+	kinematicsOptions.GoalRadiusMM = math.Min(motionCfg.PlanDeviationM*1000, 3000)
```

Comment:
Magic number should become a default const

Reply:
We dont need this, just removed it